### PR TITLE
452 delete discard button

### PR DESCRIPTION
**Pull request by fenekku**
closes https://github.com/inveniosoftware/invenio-app-rdm/issues/452

### CODE REVIEW

File: src/lib/state/actions/deposit.js [hunk 1]
```diff
@@ -56,3 +56,23 @@ export const submitFormData = (record, formik) => {
...
+ * Returns the async function that controls draft deletion.
+ *
+ * This function is different from the save/publish above because this thunk
+ * is independent of form submission.
+ *
+ * @param {object} event - click event
+ * @param {object} formik - formik object
+ */
+export const discard = (event, formik) => {
+  return async (dispatch, getState, extra) => {
```
Comment (zzacharo) [inline src/lib/state/actions/deposit.js:70]: The 3rd parameter was called `config` as it includes configuration specific to the deposit module e.g the controller class...For consistency, I would keep the `config` name...

**Review by zzacharo — APPROVED** [review]
You can merge after addressing the comments!

File: src/lib/DepositApiClient.js [hunk 2]
```diff
@@ -37,24 +37,16 @@ export class DepositApiClient {
...
-        { headers: { 'Content-Type': 'application/json' } }
-      );
+      let response = await axios_call();
       return new DepositApiClientResponse(
         response.data,  // exclude errors?
         response.data.errors,
         response.status
       );
     } catch (error) {
+      console.dir("error", error);
```
[the remark this answers is not in the document]
Comment (fenekku) [reply src/lib/DepositApiClient.js:49]: yep! Thanks!

File: src/lib/state/actions/deposit.js (hunk 1 above)
Comment (fenekku) [reply src/lib/state/actions/deposit.js:70]: This code is what makes it very confusing: https://github.com/inveniosoftware/react-invenio-deposit/blob/master/src/lib/store.js#L52 : we pass an `appConfig` to the store that holds a `config` variable and an `apiConfig` except the `apiConfig` is not just an API configuration it is also a controller and a serializer and the `config` variable (accessible by `getState().deposit.config`) well... sounds a lot like the deposit config but is different :sweat_smile: .  

What I want to do in a next PR is to replace the `apiConfig` by `controller` (controller instance) and have it hold the serializer and api client.

To be fair, I don't know of a "good approach" to divide up our code in react/redux (should we be using a controller like we are in the way we are?). I think we may have a better perspective over time when some of the semantic is cleaned up. There must also exist good examples out there (at the end of the day this is typical JS app stuff we are doing right?).

**Pull request merged**